joern-cli/frontends/rubysrc2cpg/src/main/scala/io/joern/rubysrc2cpg/parser/RubyNodeCreator.scala

```diff
@@ -548,7 +558,10 @@ class RubyNodeCreator extends RubyParserBaseVisitor[RubyNode] {
       )(ctx.toTextSpan)
     }
 
-    if (hasBlock) {
+    if (hasBlock && ctx.methodName().getText == "new") {
```

Comment:
I'm wondering if we shouldn't check if it's `Class.new` exactly, as there might be other methods named `new`

Reply:
Yeah good point